### ISSUE TITLE
Can't get PubSub object unless some other Redis command has been run first

### ISSUE THREAD

**Issue by coandco** [alisaifee/coredis#88]
In this minimal example:
```
import asyncio
from coredis import RedisCluster
from coredis.typing import Node


async def start(rclient):
    pubsub = rclient.pubsub(ignore_subscribe_messages=True)
    await pubsub.subscribe("test")


if __name__ == "__main__":
    nodes = [
        {"host": "redis1", "port": 6379},
        {"host": "redis2", "port": 6379},
        {"host": "redis3", "port": 6379}
    ]
    client = RedisCluster(startup_nodes=[Node(**x) for x in nodes])
    asyncio.run(start(client))
```

The line `await pubsub.subscribe("test")` fails with `coredis.exceptions.RedisClusterException: Cant reach a single startup node.`.  However, if I change the `start` function to look like this:

```
async def start(rclient):
    await rclient.get("foo")
    pubsub = rclient.pubsub(ignore_subscribe_messages=True)
    await pubsub.subscribe("test")
```

The pubsub object behaves as expected.  Is there some sort of initialization that's being done when the first Redis command is run?

**Comment by alisaifee** [comment]
Yes, the connection pool is initialized when the first command is awaited with the `RedisCluster` client instance. This is technically still a bug since `pubsub.subscribe` is in your example 'the first command'. This only shows up with cluster since the initialization is necessary to discover the cluster topology.

both `Redis` & `RedisCluster` clients are also awaitable  (calling await on the instance will initialize if necessary) and support the syntax:

```python
client = await RedisCluster(....)
```

or:
```python
async def do_something(client):
   await client
   pubsub = client.pubsub(....)
```

The above would be the cleaner way to resolve the issue you're having at the moment.
The pattern is unfortunately somewhat inconsistently demonstrated in the documentation (I'll get to it...).

**Comment by alisaifee** [comment]
This is resolved by #91  and the fix will be available in `4.5.2`.

Side note: if you are using pubsub with redis cluster and your redis server is >= 7.0 - support for sharded pub sub is also available ([reference](https://coredis.readthedocs.io/en/latest/handbook/pubsub.html#sharded-pub-sub)) which does not suffer from the hacky approach used for supporting cluster pubsub

**Comment by coandco** [comment]
Nice!  I'm working on getting our Redis servers upgraded to 7.x now, so hopefully I'll be able to switch to sharded pubsub in the near future.
